tests/firedrake/slate/test_slate_hybridization.py

```diff
@@ -462,3 +462,180 @@ def test_mixed_poisson_approximated_schur_jacobi_prec(setup_poisson):
 
     assert sigma_err < 1e-8
     assert u_err < 1e-8
+
+
+def test_slate_hybridization_gusto():
```

Comment:
This seems like a very large test to be adding to our test suite when the other test we have actually validates the changes. Why is this being included?

Reply:
Is it large in terms of run time or number of lines? It should work on a lower resolution mesh if that helps. I requested this test be added because it's a common solver strategy in Gusto and seems to be fragile to Firedrake changes so it would be great to have in to pick up issues and give us a starting point to work from.

Reply:
I think integrating tests for downstream packages into our CI is a really sensible idea. I think I would just like to do it in a more structured way like:
1. Adding a separate step in our CI that would clone Gusto (and others) and run a handful of their tests. This is what loopy do for us.
2. Adding a `tests/downstream` folder to contain this sort of test

Personally I prefer option 1.

Reply:
Oooh I **love** option 1!

Reply:
That's definitely useful as we want to see whether the proposed changes in the PR break things. The container won't be updated until after merging to `master`.

Reply:
I will have a stab at doing this this week. @jshipton to start with do you have a couple of tests that are both quick to run and cover a lot of Gusto's codebase? Eventually it would be cool to have a `make check` like we do.

Reply:
Are we happy to merge this then? And revisit where downstream-relevant tests should live later?

Reply:
I don't mind. Ultimately I don't think this test should be here so we can either merge it and I will remove it in my later PR, or we remove it from this PR.

Reply:
In my completely unbiased opinion, I think it should go in and be removed later... ;-)

Reply:
@connorjward I will get back to you with a set of tests tomorrow...

Reply:
In the meantime, we'd really love for this PR to go in so that Gusto main works again! Preferably with the test, but take it out if it's a barrier to merging. Thanks so much for your work on this!

Reply:
The test size has been reduced a little (see my suggestions below), so for this PR I'm going to mark this resolved and we can revisit how best to test downstream separately.

Reply:
https://github.com/firedrakeproject/firedrake/pull/4017